File: benchmarks/io/batch_write/Zig/batch_write.zig
const std = @import("std");

pub fn main() !void {
    var data: [1024]u8 = undefined;
    @memset(&data, 0);

    for (0..1000) |i| {
        var fname: [64]u8 = undefined;
        const written = try std.fmt.bufPrint(&fname, "data/smallfile_{d}.bin", .{i});
        const file = try std.fs.cwd().createFile(written, .{});
        defer file.close();
        try file.writeAll(&data);
    }
}

File: benchmarks/io/batch_write/Go/batch_write.go
package main
import (
    "fmt"
    "os"
)

func main() {
    data := make([]byte, 1024)
    for i := 0; i < 1000; i++ {
        fname := fmt.Sprintf("data/smallfile_%d.bin", i)
        f, _ := os.Create(fname)
        f.Write(data)
        f.Close()
    }
}
